plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -201,16 +201,17 @@ private List<Slice> rewriteFile(Path sourcePath, FileDeletion deletion)
         }
     }
 
-    private FileWriter createParquetFileWriter(FileSystem fileSystem, Path path, List<DeltaLakeColumnHandle> dataColumns)
+    private FileWriter createParquetFileWriter(String path, List<DeltaLakeColumnHandle> dataColumns)
     {
         ParquetWriterOptions parquetWriterOptions = ParquetWriterOptions.builder()
                 .setMaxBlockSize(getParquetWriterBlockSize(session))
                 .setMaxPageSize(getParquetWriterPageSize(session))
                 .build();
         CompressionCodecName compressionCodecName = getCompressionCodec(session).getParquetCompressionCodec();
+        TrinoFileSystem fileSystem = fileSystemFactory.create(session);
```

Comment:
Rather than creating file system over again, please put it in `DeltaLakeMergeSink` as a field instead of `fileSystemFactory`.

The same applies to other classes like `DeltaLakeUpdatablePageSource`, `HiveWriterFactory`